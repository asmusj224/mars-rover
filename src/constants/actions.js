export const REQUEST_IMAGE_GALLERY = 'REQUEST_IMAGE_GALLERY';
export const RECEIVE_IMAGE_GALLERY = 'RECEIVE_IMAGE_GALLERY';
export const RECEIVE_IMAGE_GALLERY_ERROR = 'RECEIVE_IMAGE_GALLERY_ERROR';

export const REQUEST_IMAGE_OF_THE_DAY = 'REQUEST_IMAGE_OF_THE_DAY';
export const RECEIVE_IMAGE_OF_THE_DAY = 'RECEIVE_IMAGE_OF_THE_DAY';
export const RECEIVE_IMAGE_OF_THE_DAY_ERROR = 'RECEIVE_IMAGE_OF_THE_DAY_ERROR';