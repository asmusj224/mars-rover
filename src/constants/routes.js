export const IMAGE_OF_THE_DAY = '/image-of-the-day';
export const IMAGE_GALLERY = '/image-gallery';
export const LANDING_PAGE = '/';
