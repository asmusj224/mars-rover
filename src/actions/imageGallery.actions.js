import { REQUEST_IMAGE_GALLERY, RECEIVE_IMAGE_GALLERY, RECEIVE_IMAGE_GALLERY_ERROR } from '../constants/actions';

export const requestImageGallery = () => ({ type: REQUEST_IMAGE_GALLERY });
export const receiveImageGallery = (payload) => ({ type: RECEIVE_IMAGE_GALLERY, payload });
export const receiveImageGalleryError = (payload) => ({ type: RECEIVE_IMAGE_GALLERY_ERROR, payload });