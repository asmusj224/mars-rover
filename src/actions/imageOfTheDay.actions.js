import { REQUEST_IMAGE_OF_THE_DAY, RECEIVE_IMAGE_OF_THE_DAY, RECEIVE_IMAGE_OF_THE_DAY_ERROR } from '../constants/actions';

export const requestImageOfTheDay = () => ({ type: REQUEST_IMAGE_OF_THE_DAY });
export const receieveImageOfTheDay = (payload) => ({ type: RECEIVE_IMAGE_OF_THE_DAY, payload });
export const receieveImageOfTheDayError = (payload) => ({ type: RECEIVE_IMAGE_OF_THE_DAY_ERROR, payload });
