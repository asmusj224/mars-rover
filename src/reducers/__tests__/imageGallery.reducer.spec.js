import reducer from '../imageGallery.reducer';
import * as actions from '../../constants/actions';

describe('Image Gallery Reducer', () => {
    it('should return intial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });
    it('should handle RECEIVE_IMAGE_GALLERY', () => {
        expect(reducer([], { type: actions.RECEIVE_IMAGE_GALLERY, payload: [] })).toMatchSnapshot();
    });
    it('should handle RECEIVE_IMAGE_GALLERY_ERROR', () => {
        expect(reducer([], { type: actions.RECEIVE_IMAGE_GALLERY_ERROR, payload: [] })).toMatchSnapshot();
    });
});