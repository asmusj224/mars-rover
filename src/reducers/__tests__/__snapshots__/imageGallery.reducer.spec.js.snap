// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery Reducer should handle RECEIVE_IMAGE_GALLERY 1`] = `
Object {
  "roverImages": Array [],
}
`;

exports[`Image Gallery Reducer should handle RECEIVE_IMAGE_GALLERY_ERROR 1`] = `
Object {
  "error": Array [],
}
`;

exports[`Image Gallery Reducer should return intial state 1`] = `Array []`;
